### ISSUE TITLE
Infinite "tx draft has expired" error

### ISSUE THREAD

**Issue by andywer** [satoshipay/solar#862]
As seen on one device during the launch party:

Error occured once and then appeared immediately again whenever the user tried to create a new payment. Looks as if either no new tx is created or the SubmissionProgress is stuck at the previous state.

**Comment by Marcel-Ebert** [comment]
I tested it with payments, trading and adding assets. Every time I force a 'Transaction draft has expired' error by exceeding the transaction timeout that we set on transaction creation, I have to close the submission progress dialog and get back to the form (trading, payment or add asset). 
So I always **have to submit** again. And I looked it up, we do create a new transaction every time the user hits submit: 
https://github.com/satoshipay/solar/blob/4264561cd69157febef641882fbb2743aa1e6731/src/components/Payment/PaymentForm.tsx#L348.

So it has to do something with the SubmissionProgress (?) but I was not able to reproduce this even once so... I don't know. 
@andywer are you able to reproduce?

**Comment by andywer** [comment]
Do we actually fetch a timestamp from horizon? Otherwise maybe it could be an inacurate user device time.

**Comment by Marcel-Ebert** [comment]
Seems like you're right about the inaccurate time of the user device.
Although we fetch the timebounds from horizon before we create a new transaction, the result seems to be based on the local user time.
(I tested it by setting the time of an android device back by a few hours and the fetched timebounds were calculated correctly but based on that modified time.)

I found [this](https://github.com/stellar/js-stellar-sdk/issues/305) related issue ticket which points out that there is a bug in the sdk.
